README.md

```diff
@@ -28,14 +28,28 @@ For more information on TKGS, the ```Tanzu Kubernetes Grid Service```, please se
 
 **TKGS** is the Tanzu Kubernetes Grid Service, a service available in vSphere with Tanzu to enable the deployment of TKG (guest) clusters in vSphere with Tanzu namespaces. Whilst TKGS is available in both deployments types of vSphere with Tanzu (NSX-T and native vSphere networking), the ability to backup TKGS guest clusters also requires the Supervisor Cluster to have the Velero Plugin for vSphere to be installed in the Supervisor Cluster. Since this is not possible with vSphere with Tanzu which uses native vSphere distributed switch networking, the Velero Plugin for vSphere is currently unable to backup and restore TKG guest clustes in these deployments.
 
+## Architecture
+
+Velero handles Kubernetes metadata during backup and retore. Velero relies on its plugin to backup and retore PVCs. Velero Plugin for vSphere includes the following components:
+* Velero vSphere Operator - a Supervisor Service that helps users install Velero and its vSphere plugin on the Supervisor Cluster; must be enabled through vSphere UI to support backup and restore on Supervisor and Guest Clusters
+* vSphere Plugin - deployed with Velero; called by Velero to backup and restore a PVC
+* Backupdriver - handles the backup and restore of PVCs; relies on the Data Mover to upload or download data
+* Data Mover - handles upload of snapshot data to object store and download of snapshot data from object store
+
+![Velero Plugin for vSphere Architecture](docs/vsphere-plugin-architecture.png)
+
+Backup and restore workflows are described in details [here](docs/backup-restore-workflows.md).
+
 ## Velero Plugin for vSphere Installation and Configuration details
 
 For details on how to use Velero Plugin for vSphere for each Kubernetes flavor, refer to the following documents:
 
-- [Vanilla Kubernetes](docs/vanilla.md)
+- [Vanilla Kubernetes or TKGm](docs/vanilla.md)
```

Comment:
Can we change texts of the following three items? For example,

- Guide in Vanilla Kubernetes or TKGm
- Guide in vSphere with Tanzu
- Guide in Tanzu Kubernetes Grid Service

Otherwise, users might get confused with links under `Kubernetes distributions`, since same texts are referring to different links.

Reply:
Sure